src/map.cpp

```diff
@@ -6910,6 +6910,113 @@ void map::restock_fruits( const tripoint &p, int time_since_last_actualize )
     }
 }
 
+void map::produce_sap( const tripoint &p, int time_since_last_actualize )
+{
+    if( time_since_last_actualize <= 0 ) {
+        return;
+    }
+
+    const ter_id tid = ter( p );
+    if( tid != t_tree_maple_tapped ) {
+        return;
+    }
+
+    // Amount of maple sap litres produced per season per tap
+    int maple_sap_per_season = 56;
+
+    // How many turns to produce 1 charge (250 ml) of sap?
+    int turns_season = DAYS( calendar::season_length() );
+    int producing_length = int( 0.75f * turns_season );
+
+    int turns_to_produce = producing_length / ( maple_sap_per_season * 4 );
```

Comment:
I'm not quite sure, but this value can probably be computed during compilation (at least all these variables can be static `const`s for sure).